app/services/listing_service/store/shape.rb

```diff
@@ -193,8 +196,19 @@ def from_unit_model_attributes(hash)
       }, hash)
   end
 
-  def find_shape_model(community_id:, listing_shape_id:)
-    find_shape_models(community_id: community_id).where(id: listing_shape_id).first
+  def find_shape_model(community_id:, listing_shape_id: nil, name: nil)
+    q = find_shape_models(community_id: community_id)
+
+    q =
+      if listing_shape_id.present?
+        q.where(id: listing_shape_id)
+      elsif name.present?
+        q.where(name: name)
+      else
+        raise ArgumentError.new("Must have either name or listing shape id present")
+      end
+
+    q.first
```

Comment:
no need for local variable q, just two branches that both do the whole chain of ops.